azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java

```diff
@@ -546,7 +555,17 @@ private void createPod(final int executionId) throws ExecutorManagerException {
     allImageTypes.addAll(jobTypes);
     final VersionSet versionSet = fetchVersionSet(executionId, flowParam, allImageTypes);
     final V1PodSpec podSpec = createPodSpec(executionId, versionSet, jobTypes, flowParam);
-
+    if (StringUtils.isNotEmpty(this.podTemplatePath)) {
+      try {
+        AzKubernetesV1PodTemplate podTemplate = AzKubernetesV1PodTemplate
+            .getInstance(this.podTemplatePath);
+        mergePodSpec(podSpec, podTemplate);
+      } catch (IOException e) {
+        logger.info("ExecId: {}, Failed to create k8s pod from template: {}", executionId,
```

Comment:
The pod creation should not fail if template is not available.

Reply:
We already have a check for the existence of the property "azkaban.kubernetes.pod.template.path". If this is not specified, then only we will try to use a template. So, if we fail to read the template after the property is specified, then we should fail as we are expecting something else.